content/articles/docker-push-for-publishing-images-to-docker-hub/index.md

```diff
@@ -0,0 +1,125 @@
+
+```
+touch Dockerfile
+```
+
+Since the file we created is empty, open it via a text editor of your choosing and update the file as shown below:
+
+```
+FROM linux
+
+MAINTAINER testUser
+
+RUN apt-get update
+
+CMD ["echo", "Welcome to Dockerfile"]
+```
+
+* **FROM** - specifies the prop of the created image. It can start from the base image or the root image.
+* **MAINTAINER** - Defines the author of that particular image. It can take a first name, last name, or email. **LABEL** attribute further highlights more about the image.
+* **RUN** - It is a command that carries the set of instructions for executing the Dockerfile while building the image.
+* **CMD** - The command provides a revert for a Dockerfile that's executing.
+
+To view the contents of the `Dockerfile` you can use the `cat` command in the terminal:
+
+```
+oruko@oruko-ThinkPad-T520:~/Documents/TestDocker$ cat Dockerfile
+FROM ubuntu
+
+MAINTAINER testUser
+
+RUN apt-get update
+
+CMD ["echo", "Welcome to Dockerfile"]
+```
+
+### Creating a repository on Docker Hub
+Now that we have created our Dockerfile before we push it to an online repository, let's create a repository within Docker.
+If you're well-acquainted with the way Github works then Docker hub isn't that different from it. So head over to [Docker Hub](https://hub.docker.com/) and register an account. After signup, click the `Repositories` tab in the navbar, you'll see a form like the one below:
+
+![repository](engineering-education/docker-push-for-publishing-images-to-docker-hub/repo.jpg)
+
+Create a repository called `docker-push` as that is the example we'll be using throughout the article. Now that our repository is set,  let's proceed and create an image from Docker and push it to the repository we created earlier.
+
+### Build Docker Image using Docker Hub and for organization
+To be able to build an image in Docker, the command below is used:
+ ```
+ docker build -t username/repository_name .
+ ```
+
+The `-t` flag helps when dealing with various images in identifying which name they belong to. The `username` is your Docker hub name and the `repository_name` in this case is `docker-push` the repository we created earlier. We add a period `.` when in the folder of the Dockerfile. With this hindsight, let's proceed and build our Docker image by executing the command changing the username with yours as it appears in the Docker hub and the repository with `docker-push` as displayed below:
+
+```
+docker build -t bullet08/docker-push .
+```
+
+For more about Docker Hub commands, check the documentation [here](https://docs.docker.com/docker-hub/repos/).
+
+The same approach is used when building Docker images for organizations, you change the username with the organization's account name and Docker hub repository of the organization.
+
+### Pushing Docker image
+Before we push the Docker image we need to log into the Docker hub. We can do this effortlessly on the command-line using the command shown below:
```

Comment:
Make sure you commas where necessary. "Before we push the Docker image, we......". I've added for you a lot of commas on my preliminary edits. Please check for reference.